File: LeetcodeQuestions/triangle-120.py
class Solution(object):
    def minimumTotal(self, triangle):
        """
        :type triangle: List[List[int]]
        :rtype: int
        """
        dp = triangle[-1]

        i = len(triangle) - 2
        while i >= 0:
            for j in range(i + 1):
                dp[j] = min(dp[j], dp[j + 1]) + triangle[i][j]
            i -= 1
        return dp[0]

File: LeetcodeQuestions/triangle-120.rb
# @param {Integer[][]} triangle
# @return {Integer}
def minimum_total(triangle)
    dp = triangle[triangle.length - 1]
    
    i = triangle.length - 2
    while i >= 0
        j = 0
        while (j <= i)
            dp[j] = [dp[j], dp[j + 1]].min + triangle[i][j]
            j += 1
        end
        i -= 1
    end
    dp[0]
        
end

